lib/matplotlib/cbook.py

```diff
@@ -2701,3 +2702,15 @@ def __exit__(self, exc_type, exc_value, traceback):
                     os.rmdir(path)
                 except OSError:
                     pass
+
+
+def is_matrix(obj):
+    '''
+    This is a test for whether the input is a matrix.
+    If the input is a matrix, raise an error. Otherwise,
+    return the object as it is.
+    '''
+    cast_result = np.asanyarray(obj)
+    if isinstance(cast_result, np.matrix):
+        raise ValueError("The input cannot be matrix")
+    return obj
```

Comment:
You'd want to return the cast objects. Ie, if X is a list, you want the returned object to be an array.